app/src/main/java/de/tum/in/tumcampusapp/component/tumui/calendar/CalendarActivity.java

```diff
@@ -406,72 +413,16 @@ public String interpretTime(int hour, int minutes) {
-        dialog.setPositiveButton(R.string.delete, (dialog1, which) -> {
-            TUMOnlineRequest<DeleteEvent> request = new TUMOnlineRequest<>(
-                    TUMOnlineConst.Companion.getDELETE_EVENT(), this, true);
-            request.setParameter("pTerminNr", nr);
-            request.fetchInteractive(this, new TUMOnlineRequestFetchListener<DeleteEvent>() {
-                @Override
-                public void onNoInternetError() {
-                    Toast.makeText(getApplicationContext(), "Error: you are not connected to the internet", Toast.LENGTH_SHORT)
-                         .show();
-                }
-
-                @Override
-                public void onFetch(DeleteEvent response) {
-                    detailsFragment.dismiss();
-                    TcaDb.getInstance(getApplicationContext())
-                         .calendarDao()
-                         .delete(nr);
-                    refreshWeekView();
-                    Toast.makeText(getApplicationContext(), R.string.delete_event_confirmation, Toast.LENGTH_SHORT)
-                         .show();
-                }
-
-                @Override
-                public void onFetchCancelled() {
-                    Toast.makeText(getApplicationContext(), R.string.something_wrong, Toast.LENGTH_SHORT)
-                         .show();
-                }
-
-                @Override
-                public void onFetchError(String errorReason) {
-                    Toast.makeText(getApplicationContext(), R.string.delete_event_error, Toast.LENGTH_LONG)
-                         .show();
-                }
-
-                @Override
-                public void onNoDataToShow() {
-                    Toast.makeText(getApplicationContext(), R.string.something_wrong, Toast.LENGTH_SHORT)
-                         .show();
-                }
-            });
-        });
-        dialog.setNegativeButton(R.string.cancel, null);
-        dialog.show();
-    }
-
     @Override
-    public void onEventClick(WeekViewEvent weekViewEvent, RectF rectF) {
-        detailsFragment = new CalendarDetailsFragment();
-        Bundle bundle = new Bundle();
-        CalendarItem item = calendarController.getCalendarItemByStartAndEndTime(
-                new DateTime(weekViewEvent.getStartTime()),
-                new DateTime(weekViewEvent.getEndTime()));
-        bundle.putString(CALENDAR_ID_PARAM, item.getNr());
-        detailsFragment.setArguments(bundle);
-        detailsFragment.show(getSupportFragmentManager(), null);
+    public void onEventDeleted(@NotNull String eventId) {
+        TcaDb.getInstance(this).calendarDao().delete(eventId);
+        refreshWeekView();
+        Utils.showToast(this, R.string.delete_event_confirmation);
     }
 
-    protected void editEvent(final CalendarItem calendarItem) {
+    @Override
+    public void onEditEvent(@NotNull CalendarItem calendarItem) {
+        // TODO: CalendarItem should implement Parcelable
```

Comment:
You still want to do this?

Reply:
That’s more of a “do this at some point” comment. People keep telling me my pull requests are too large 🙈